controllers/loadtest_controller.go

```diff
@@ -153,7 +198,18 @@ func newDriverPod(loadtest *grpcv1.LoadTest, component *grpcv1.Component) (*core
 		return nil, err
 	}
 
-	addDriverPort(&pod.Spec.Containers[0])
+	spec := &pod.Spec
+
+	// TODO: Avoid referencing containers by index, use names
+	addDriverPort(&spec.Containers[0])
+
+	// TODO: Handle more than 1 scenario
+
+	// At this point, we assume validation will have caught any loadtest
+	// without a scenario.
```

Comment:
We need to add this logic to the controller itself. I don't rely on returning and propagating an error here, because we should not wait until creating the driver to see there is no scenario on the load test. We should mark it as an error on the very first reconcile before creating anything.